client/ssh.go

```
@@ -185,16 +215,26 @@ func (c *SSHClient) connect() error {
 		return nil
 	}
 
-	signer, err := parsePrivateKey(c.config.PrivateKey)
-	if err != nil {
-		return err
+	var authMethods []ssh.AuthMethod
+
+	if c.config.UseAgent {
+		conn, err := net.Dial("unix", c.config.AgentSocket)
```

golangci-lint (2.12.2)

[error] net.Dial must not be called. use (*net.Dialer).DialContext

(noctx)